### DISCUSSION TITLE
Error P1002 at remote connection

### DISCUSSION THREAD

**Discussion by rondered** [prisma/prisma#5816] [Q&A]
Hi, we use the same configuration that we have used in TypeORM (same localhost, port etc..) and now we are moving to Prisma but on remote environment where we deploy the app (everything works fine locally) we get this error:
`> pii-api@1.0.0 start:nobuild /app
> npm run prisma:migration && npm run prisma:seed && node ./dist/main
> pii-api@1.0.0 prisma:migration /app
> npm run prisma:cli -- migrate deploy --preview-feature
> pii-api@1.0.0 prisma:cli /app
> node ./node_modules/.bin/prisma "migrate" "deploy" "--preview-feature"
Environment variables loaded from .env
Environment variables loaded from prisma/.env
Prisma schema loaded from schema.prisma
Datasource "db": PostgreSQL database "oct_dev", schema "pii" at "pgpool.pg:5432"
1 migration found in prisma/migrations
Error: P1002
The database server at `pgpool.pg`:`5432` was reached but timed out.
Please try again.
Please make sure your database server is running at `pgpool.pg`:`5432`.
npm ERR! code ELIFECYCLE
npm ERR! errno 1`

We want the app to migrate, then seed, then run everytime it starts and looks like migration is failing.

**Comment by ruheni** [comment]
Hi @sronmiz 👋🏽

Could you share a few more details for context, such as the scripts you have in `package.json` and your deployment provider? 🙂

**Reply by Jolg42** [reply]
The discussion is actually happening in public slack here https://prisma.slack.com/archives/CA491RJH0/p1614161039079900

**Reply by rondered** [reply]
Hey guys, thanks for your help.
so - the scripts are:
`    "prisma:cli": "npm run build && node ./node_modules/.bin/prisma",
    "prisma:migration": "npm run prisma:cli -- migrate deploy --preview-feature",
    "prisma:seed": "npm run prisma:cli -- db seed --preview-feature",`
   
Me and Julius ran the same connection string inside Rust engine and found out that sometimes the query "SELECT 1" returns an error because the database disconnected, supported by a log thrown by the engine.
I have tried to "hardcode" the connection string inside the provider in the schema and had no success too - so there is no way I try to connect to "different" connection string.
Looks like different js orms have some kind of reconnecting mechanism which the engine doesn't have.
Unfortuanly I can't provide the team access to our development database at all.

**Reply by janpio** [reply]
Not sure if this already happened, but if you fundamentally understand what is going on, it would be ace if you could create an issue here in this repo so this is documented and can (potentially) be looked at.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
